src/users/users.service.ts

```diff
@@ -58,15 +59,51 @@ export class UsersService {
         await this.loginRepo.delete({employee: user});
         return await this.usersRepository.delete(user);
     }
-    
+    // TEMPORARY ONLY
+    // Create Dummy if Database is empty.
+    // This endpoint will add admin Dummy
+    async createDummy(): Promise<Users> {
+        const user = new Users();
+        user.role = Role.Admin;
+        user.employeeId = 0;
+        user.firstName = 'dummy';
+        user.lastName = 'dummy';
+        user.isManager = true;
+        user.positionTitle = 'dummy';
+        user.startDate = new Date("2014-12-18");
+
+        let company = await this.companyservice.createCompany({
+            companyId: 1, 
+            name: 'dummy', 
+            tags: undefined, recognitions: undefined,
+            users: undefined
+        });
+        user.company = company
+
+        const login = new Login();
+        login.email = 'dummy';
+        login.password = 'dummy';
+        login.employee = await this.usersRepository.save(user);
+        await this.loginRepo.save(login);
+        return user
+    }
+
     async createUser(createuserDto: Users & Login & {managerId: number} & {companyName: string}): Promise<Users> {    
         const user = new Users();
         if (createuserDto.company != undefined) {
             user.company = createuserDto.company;
+
         }
         else{
             if (createuserDto.companyId != undefined) {
                 let company = await this.companyRepository.findOne({where:{companyId: createuserDto.companyId}})
+                // If company.name to companyName if they are not the same
+                // Don't 
+                if (company.name != createuserDto.companyName){
+                    company.name = createuserDto.companyName; 
```

Comment:
Shouldn't this be the other way around? In the current situation, if the new user object has a misspelling of the company name, then the company name will become misspelled in the database. The createuserDto.companyName should be changed instead of company.name.

Reply:
Also, if we are creating a user, we might only want to specify the companyID and not the company name. So maybe we should just return an error if the company ID isn't found.

Reply:
the sample data contains company names and companyId at the same time. So if companyId is not provided, then we can search DB for company name. 
The changing company name is my way to circumvent my Dummy user. Dummy user's company name is 'dummy'. This is temporary.  I will change it when we know how to deal with  empty DB